tests/main/interfaces-microstack-support/task.yaml

```diff
@@ -0,0 +1,34 @@
+summary: Ensure that the microstack-support device delegation works.
+
+details: |
+    The microstack-support interface delegates device cgroup handling to the
+    snap.
+
+prepare: |
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-sh
+
+execute: |
+    echo "When the interface is connected"
+    snap connect test-snapd-sh:microstack-support
+    snap connect test-snapd-sh:hardware-observe
+
+    # Start a udevadm monitor inside the snap, in the background
+    echo "Start monitoring udev events"
+    test-snapd-sh.cmd udevadm monitor > monitor.log &
+    pid=$!
+    # cleanup the background job on exit
+    trap 'kill $pid; wait $pid' EXIT
```

Comment:
I think a more appropriate test here would be to check that the process that is part of a daemon in this snap has no effective cgroup enforced, i.e. either it has a cgroup with all devices allowed or the process is not in a snapd managed cgroup at all (I think which specific one depends on whether we are on cgroup v1 or v2).

Probably something like checking in `/proc/<pid>/cgroup` for the devices line and then checking what devices.list says for that cgroup in sysfs, but the details may prove slightly complicated handling both cgroupsv1 and cgroupsv2